torch/utils/hooks.py

```diff
@@ -83,7 +83,7 @@ def warn_if_has_hooks(tensor):
     if tensor._backward_hooks:
         for k in tensor._backward_hooks:
             hook = tensor._backward_hooks[k]
-            if not hasattr(k, "__torch_unserializable__"):
+            if not hasattr(hook, "__torch_unserializable__"):
                 warnings.warn(f"backward hook {repr(hook)} on tensor will not be "
                               "serialized.  If this is expected, you can "
                               "decorate the function with @torch.utils.hooks.unserializable_hook "
```

Comment:
cc: @ezyang since I think you wrote this code (6 years ago 😄 )

This change looks right to me, we should be looking for `__torch_unserializable__` on the `hook` function, not the key, which looks to be an `int`:
https://github.com/pytorch/pytorch/blob/a988b4ed76ca6ce43a7e18c2e7976fdbce210375/torch/_tensor.py#L566-L571

Reply:
WHOOPS. LGTM!